sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_responses.hpp

```diff
@@ -87,7 +87,7 @@ namespace Azure { namespace Storage { namespace Blobs {
      * @return A reference to an #Azure::Core::Http::RawResponse.
      * @note Does not give up ownership of the RawResponse.
      */
-    Azure::Core::Http::RawResponse const& GetRawResponse() const override { return *m_rawResponse; }
+    Azure::Core::Http::RawResponse const& GetRawResponseInternal() const override { return *m_rawResponse; }
```

Comment:
why do we need this? We already have an`Azure::Core::Http::RawResponse const& GetRawResponse() const` inherited from base class. Can we just remove this function?

Reply:
GetRawResponse() is no longer virtual.  It validates that the internal raw_response is not null before returning the reference of the raw response. 
Since m_rawResponse is protected, any derived class can move it out of the class and run into a seg fault later when calling GetRawResponse.

Reply:
If we don't allow move raw response from `Operation<T>`, only allow copy, we don't have this problem anymore and we don't need this `GetRawResponseInternal`.

Reply:
We do still.
```C++
MyCustomOperation option = MyCustomOperation();
option.GetRawResponse(); // What would this do?
```

Reply:
>
```
MyCustomOperation option = MyCustomOperation();
option.GetRawResponse(); // What would this do?
```

I don't see any problem in this piece of code.
`Operation<T>` is movable and non-copiable. `MyCustomOperation()` returns an rvalue, then it's moved into `option`. Now `option` has the raw response. `GetRawResponse` returns the raw response inside `option`

Reply:
> we don't have this problem anymore and we don't need this `GetRawResponseInternal`.

I am addressing this part, not about the copy/move conversation.

The purpose of `GetRawResponseInternal` is so that we can throw a run-time error in the above code I shared (because there is no raw response to return).

Otherwise, everyone who implements `GetRawResponse()` would need to guard against an empty response. If the response is not set, how can we return a reference to it?